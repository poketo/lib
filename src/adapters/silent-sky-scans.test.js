import poketo from '../index';
import adapter from './silent-sky-scans';

describe('SilentSkyScans', () => {
  const server = new AdapterVcrServer(adapter);

  beforeAll(async () => {
    await server.listenAndMock(57169);
  });

  afterAll(() => {
    server.close();
  });

  describe('getSeries', () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'silent-sky-scans:all_rounder_meguru_',
      );

      expect(metadata).toMatchSnapshot();

      const chapterNumbersToTest = ['64', '81'];
      const chaptersToTest = chapters.filter(chapter =>
        chapterNumbersToTest.includes(chapter.chapterNumber),
      );
      expect(chaptersToTest).toMatchSnapshot();
    });
  });
});
