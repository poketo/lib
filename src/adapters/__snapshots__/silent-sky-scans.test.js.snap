// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SilentSkyScans getSeries returns a metadata object 1`] = `
Object {
  "author": null,
  "coverImageUrl": "http://reader.silentsky-scans.net/content/comics/all_rounder_meguru__51c584cb54c37/thumb_v12.jpg",
  "description": "Meguru joins a mixed martial arts gym hoping to become stronger. 
He meets an old childhood friend who is practicing the same kind of shooto martial arts, and who harbors a dark secret related to their past.",
  "id": "silent-sky-scans:all_rounder_meguru_",
  "site": Object {
    "id": "silent-sky-scans",
    "name": "Silent Sky Scans",
  },
  "slug": "all_rounder_meguru_",
  "status": "UNKNOWN",
  "supportsReading": true,
  "title": "All Rounder Meguru",
  "updatedAt": Any<Number>,
  "url": "http://localhost:57169/series/all_rounder_meguru_",
}
`;

exports[`SilentSkyScans getSeries returns a metadata object 2`] = `
Array [
  Object {
    "chapterNumber": "81",
    "createdAt": 1394828845,
    "id": "silent-sky-scans:all_rounder_meguru_:en/9/81",
    "slug": "en/9/81",
    "title": "Keiji Mitsuya",
    "url": "http://localhost:57169/read/all_rounder_meguru_/en/9/81/page/1",
    "volumeNumber": "9",
  },
  Object {
    "chapterNumber": "64",
    "createdAt": 1371899118,
    "id": "silent-sky-scans:all_rounder_meguru_:en/7/64",
    "slug": "en/7/64",
    "title": "Ten Years, Part 2",
    "url": "http://localhost:57169/read/all_rounder_meguru_/en/7/64/page/1",
    "volumeNumber": "7",
  },
]
`;
